### PR TITLE
Feat: datadog defaults

### PR DESCRIPTION
**Pull request by DCamma**

### CODE REVIEW

File: modules/datadog/main.tf [hunk 1]
```diff
@@ -157,15 +162,11 @@ resource "helm_release" "datadog_agent" {
                         name: datadog-keys
                         key: app-key
                   - name: DD_LOGS_ENABLED
-                    value: "false"
-                  - name: DD_ENV
-                    value: "${var.environment}"
-                  - name: DD_CLUSTER_NAME
-                    value: "${var.cluster_name}"
```
Comment (DCamma) [inline modules/datadog/main.tf:164]: Test: remove this and see if the global settings are enough

File: modules/datadog/variables.tf [hunk 2]
```diff
@@ -63,3 +63,8 @@ variable "datadog_agent_version_fargate" {
   type        = string
   default     = "7.54.0"
 }
+
+variable "product_name" {
+  description = "Name of the product"
+  type        = string
+}
```
Comment (DCamma) [inline modules/datadog/variables.tf:70]: Without default this is a required var opinions on a possible default? used in [modules/datadog/main.tfR121

File: modules/datadog/variables.tf [hunk 3]
```diff
@@ -63,3 +63,8 @@ variable "datadog_agent_version_fargate" {
   type        = string
   default     = "7.54.0"
 }
+
+variable "product_name" {
```
Comment (DCamma) [inline modules/datadog/variables.tf:67]: I would prefer team_name, but I put product to align with the currently used tag in datadog (e.g. in disco pmd sfmc titan etc we use product:<teamname> so it felt more accurate calling the var product as well) 
I don't have a strong opinion tho

File: modules/datadog/main.tf [hunk 4]
```diff
@@ -158,14 +163,12 @@ resource "helm_release" "datadog_agent" {
                         key: app-key
                   - name: DD_LOGS_ENABLED
                     value: "false"
-                  - name: DD_ENV
-                    value: "${var.environment}"
-                  - name: DD_CLUSTER_NAME
-                    value: "${var.cluster_name}"
-              selectors:
+                  - name: DD_TAGS
+                    value: "env:${var.environment}-${var.product_name}"
```
Comment (DCamma) [inline modules/datadog/main.tf:167]: Wasn't able to set more than one tag: DD_TAGS cannot be replicated, using a comma doesn't work and the two tags are concatenated with an underscore, having a space between them doesn't work either (again becomes an underscore). DD_TAGS only accepts a string, not a list

File: modules/datadog/main.tf [hunk 5]
```diff
@@ -158,14 +163,12 @@ resource "helm_release" "datadog_agent" {
                         key: app-key
                   - name: DD_LOGS_ENABLED
                     value: "false"
-                  - name: DD_ENV
-                    value: "${var.environment}"
-                  - name: DD_CLUSTER_NAME
-                    value: "${var.cluster_name}"
```
Comment (DCamma) [inline modules/datadog/main.tf:164]: Not needed as this one is actually inherited

File: modules/datadog/main.tf (hunk 1 above)
Comment (DCamma) [reply modules/datadog/main.tf]: works for dd_cluster_name but not for dd_env

File: modules/datadog/main.tf [hunk 6]
```diff
@@ -210,9 +213,12 @@ resource "helm_release" "datadog_agent" {
...
 # Delays the annotations until the Datadog Agent is ready
 resource "time_sleep" "this" {
-  depends_on = [helm_release.datadog_agent]
+  create_duration = "60s"
+  triggers = {
+    helm_values = sha256(join("", helm_release.datadog_agent.values))
+  }
 
-  create_duration = "120s"
+  depends_on = [helm_release.datadog_agent]
```
Comment (AlfGot) [inline modules/datadog/main.tf:221]: I'm not sure the depends_on is still needed.

File: modules/datadog/variables.tf (hunk 2 above)
Comment (AlfGot) [reply modules/datadog/variables.tf:70]: what about _example_ ?

**Review by AlfGot — APPROVED** [review]

**Comment by swibrow** [comment]
@DCamma this is a feat since you add a var and change some defaults

File: modules/datadog/variables.tf (hunk 2 above)
Comment (DCamma) [reply modules/datadog/variables.tf:70]: But I prefer not setting it so that the user is required to set something meaningful.

**Pull request merged**